api/src/BcGov.Malt.Web/Services/SharepointResourceUserManagementService.cs

```diff
@@ -1,28 +1,308 @@
-﻿using System.Threading.Tasks;
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
 using BcGov.Malt.Web.Models.Configuration;
+using BcGov.Malt.Web.Models.SharePoint;
+using BcGov.Malt.Web.Services.Sharepoint;
+using Microsoft.Extensions.Logging;
+using Refit;
 
 namespace BcGov.Malt.Web.Services
 {
-    public class SharepointResourceUserManagementService : ResourceUserManagementService
+
+    public interface ISharePointClient
+    {
+        [Get("/_api/Web?$select=Title,ServerRelativeUrl")]
+        public Task<GetSharePointWebVerboseResponse> GetWebAsync();
+
+        [Get("/_api/Web/SiteGroups?$select=Id,Title")]
+        public Task<GetSiteGroupsVerboseResponse> GetSiteGroupsAsync();
+
+        [Get("/_api/Web/SiteGroups/GetById({siteGroupId})/Users")]
+        public Task<string> GetSiteGroupUsersAsync(int siteGroupId);
+
+        [Get("/_api/Web/SiteGroups?$filter=Title%20eq%20'{title}'")]
+        public Task<GetSiteGroupsVerboseResponse> GetSiteGroupsByTitleAsync(string title);
+
+        [Get("/_api/Web/GetUserById({userId})/Groups")]
+        public Task<GetSiteGroupsVerboseResponse> GetUserGroupsAsync(int userId);
+
+        /////// <remarks>Current version of Refit always encodes the loginName making this not working</remarks>>
+        ////[Get("/_api/Web/SiteGroups/GetById({siteGroupId})/Users?$filter=LoginName eq '{loginName}'")]
+        ////public Task<GetSiteUsersVerboseResponse> GetUserInGroupByLoginNameAsync(int siteGroupId, string loginName);

```

Comment:
can this be removed?

Reply:
I want to leave it in so we know why we are not filtering by login name like OData supports. There is a pending change in that will allow us not to escape parameter and make the overall code more efficient.